web/src/app/services/data-store/data-store.service.ts

```diff
@@ -77,43 +78,26 @@ export class DataStoreService {
    * @param id the id of the requested survey.
    */
   loadSurvey$(id: string) {
-    return this.db
-      .collection(SURVEYS_COLLECTION_NAME)
-      .doc(id)
-      .valueChanges()
-      .pipe(
-        // Convert object to Survey instance.
-        map(data => FirebaseDataConverter.toSurvey(id, data as DocumentData))
-      );
-  }
+    return combineLatest([
+      this.db.collection(SURVEYS_COLLECTION_NAME).doc(id).valueChanges(),
+      this.db
+        .collection(`${SURVEYS_COLLECTION_NAME}/${id}/jobs`)
+        .valueChanges(),
+    ]).pipe(
+      switchMap(surveyAndJobs => {
+        const [s, jobs] = surveyAndJobs;
 
-  /**
-   * Returns an Observable that loads and emits the job with the specified
-   * uuid and survey uuid.
-   *
-   * @param jobId the id of the requested job.
-   * @param surveyId the id of the requested survey.
-   */
-  loadJob$(jobId: string, surveyId: string) {
-    return this.db
-      .collection(SURVEYS_COLLECTION_NAME)
-      .doc(surveyId)
-      .valueChanges()
-      .pipe(
-        // Convert object to Survey instance.
-        map(data => {
-          const job = FirebaseDataConverter.toSurvey(
-            surveyId,
-            data as DocumentData
-          ).getJob(jobId);
-
-          if (!job) {
-            throw Error(`Job $jobId not found`);
-          }
-
-          return job;
-        })
-      );
+        const survey = surveyDocToModel(
+          id,
+          s as DocumentData,
+          jobs as DocumentData[]
```

Comment:
This seems like the wrong place to combine the data.. can we jus t do the conversion here and not below?